diff --git a/benchspec/CPU2006/464.h264ref/src/context_ini.c b/benchspec/CPU2006/464.h264ref/src/context_ini.c
index 90cb0f00..9560602e 100644
--- a/benchspec/CPU2006/464.h264ref/src/context_ini.c
+++ b/benchspec/CPU2006/464.h264ref/src/context_ini.c
@@ -219,13 +219,13 @@ void init_contexts ()
   BIARI_CTX_INIT1 (                 NUM_IPR_CTX,  tc->ipr_contexts,     INIT_IPR,       img->model_number);
   BIARI_CTX_INIT1 (                 NUM_CIPR_CTX, tc->cipr_contexts,    INIT_CIPR,      img->model_number);
   BIARI_CTX_INIT2 (3,               NUM_CBP_CTX,  tc->cbp_contexts,     INIT_CBP,       img->model_number);
-  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES, NUM_BCBP_CTX, tc->bcbp_contexts,    INIT_BCBP,      img->model_number);
-  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES, NUM_MAP_CTX,  tc->map_contexts,     INIT_MAP,       img->model_number);
-  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES, NUM_LAST_CTX, tc->last_contexts,    INIT_LAST,      img->model_number);
-  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES, NUM_ONE_CTX,  tc->one_contexts,     INIT_ONE,       img->model_number);
-  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES, NUM_ABS_CTX,  tc->abs_contexts,     INIT_ABS,       img->model_number);
-  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES, NUM_MAP_CTX,  tc->fld_map_contexts, INIT_FLD_MAP,   img->model_number);
-  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES, NUM_LAST_CTX, tc->fld_last_contexts,INIT_FLD_LAST,  img->model_number);
+  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES-2, NUM_BCBP_CTX, tc->bcbp_contexts,    INIT_BCBP,      img->model_number);
+  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES-2, NUM_MAP_CTX,  tc->map_contexts,     INIT_MAP,       img->model_number);
+  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES-2, NUM_LAST_CTX, tc->last_contexts,    INIT_LAST,      img->model_number);
+  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES-2, NUM_ONE_CTX,  tc->one_contexts,     INIT_ONE,       img->model_number);
+  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES-2, NUM_ABS_CTX,  tc->abs_contexts,     INIT_ABS,       img->model_number);
+  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES-2, NUM_MAP_CTX,  tc->fld_map_contexts, INIT_FLD_MAP,   img->model_number);
+  BIARI_CTX_INIT2 (NUM_BLOCK_TYPES-2, NUM_LAST_CTX, tc->fld_last_contexts,INIT_FLD_LAST,  img->model_number);
 }


diff --git a/benchspec/CPU2006/464.h264ref/src/mv-search.c b/benchspec/CPU2006/464.h264ref/src/mv-search.c
index 504d8c4b..5e42c7af 100644
--- a/benchspec/CPU2006/464.h264ref/src/mv-search.c
+++ b/benchspec/CPU2006/464.h264ref/src/mv-search.c
@@ -1016,7 +1016,7 @@ FastFullPelBlockMotionSearch (pel_t**   orig_pic,     // <--  not used
 int
 SATD (int* diff, int use_hadamard)
 {
-  int k, satd = 0, m[16], dd, d[16];
+  int k, satd = 0, m[16], dd, d[17];

   if (use_hadamard)
   {
